src/content/docs/game-design/godot/2dPlatformerGame.mdx

```diff
@@ -0,0 +1,231 @@
+
+![Screenshot showing TileSet and Tilemap menus](/src/assets/godot/2DPlatformer/screenshot6.png)
+
+Inside the TileSet menu, click the + button and select "Atlas". Navigate to the world_tileset image in sprites and load that. You will be asked if Godot can automatically create tiles, select yes. Now, click the "TileSet" button in the inspector, go to the "Physics Layers" drop down and click the "Add Element" button. 
+
+![Screenshot showing physics layer add element](/src/assets/godot/2DPlatformer/screenshot7.png)
+
+Now back to the TileSet menu, click the Paint button, click the drop down that appears and select "Physics Layer 0". Now click on some of the square platforms and they should be given a blue collision box. It will look highlighted on the Base Tiles sections. 
+
+![Screenshot showing TileMap collision boxes](/src/assets/godot/2DPlatformer/screenshot8.png)
+
+When you click on a sprite, you should see it appear in the paint window. You can then drag the white diamonds so that the blue collision box roughly covers the sprite. For things like the ground sprites, the default square is ok, but you may need to play around for other sprites, such as the bridge sprites.
+
+Now open the TileMap menu at the bottom of the screen and click on a tile to enable drawing in the editor. You can click to place individual tiles, or click and drag to paint them following the cursor. Play around with the tile paint tools to create a fun level to play. Some cool things to check out are the Rect tool and the Place Random Tile options.
+
+Now add a new TileMapLayer as a child of Level 1 and place it above Foreground in the scene tree. 
+
+![Screenshot showing scene tree](/src/assets/godot/2DPlatformer/screenshot9.png)
+
+Repeat the steps used for the Foreground (ignoring the physics steps) to create your Background. If you placed the Background TileMapLayer above the Foreground, Background tiles should appear behind the Foreground tiles. Unless your player is below the TileMapLayers, they will disappear behind the background, so make sure to move them below the Foreground in the scene tree.
+
+## Adding Movement
+
+Now we are going to add gravity and movement to the player. Go back to your player scene and attach a script by right clicking the player node and clicking "attach script". Click "create" as the defaults are what we want. Now if you run your game, you should be able to run around. You may want to reduce your speed and jump height, depending on how you want the game to feel (I recommend a speed of 150 and a jump velocity of -300). You may notice that your player doesn't flip when going to the left, nor does the animation change from idle to run when you're moving. To solve this, add the below code above `move_and_slide()` as well as adding `@onready var sprite_node = $AnimatedSprite2D` below `extends CharacterBody2D`. Note that the animation names are case sensitive, so if your player disappears that may be why.
+
+```gdscript
```

Comment:
This should show where it is in context of the wider codebase (can use a `...` to show lots of code with the line before visible. The code should also be indented the correct amount.